api/src/main/java/org/pmiops/workbench/db/dao/UserRecentResourceServiceImpl.java

```diff
@@ -117,7 +120,16 @@ public void deleteNotebookEntry(long workspaceId, long userId, String notebookPa
    */
   @Override
   public List<DbUserRecentResource> findAllResourcesByUser(long userId) {
-    return userRecentResourceDao.findUserRecentResourcesByUserIdOrderByLastAccessDateDesc(userId);
+    try {
+      return DbRetryUtils.executeAndRetry(
+          () ->
+              userRecentResourceDao.findUserRecentResourcesByUserIdOrderByLastAccessDateDesc(
+                  userId),
+          Duration.ofSeconds(1),
+          5);
+    } catch (InterruptedException e) {
```

Comment:
should we also have a catch for dataacessexception, in case it failed 5 times? Its not likely but in case it does

Reply:
dataacessexception is not a checked exception, I think we can just throw whatever we get from substream code.